registry/repositories.go

```diff
@@ -232,16 +278,28 @@ func streamHarborProjectsPage(ctx context.Context, registry *Registry, c chan st
 			continue
 		}
 
+		var harborProjRepo string = ctx.Value("harborProjRepo").(string)
+
+		if strings.Contains(harborProjRepo, "v2.0") {
+			ctx = context.WithValue(ctx, "harborProjRepo", (harborProjRepo + project.Name + "/repositories"))
+		} else {
+			ctx = context.WithValue(ctx, "harborProjRepo", (harborProjRepo + string(project.ID)))
+		}
+
 		if !streamHarborProjectRepos(ctx, project, registry, c, e) {
 			return false
 		}
+		ctx = context.WithValue(ctx, "harborProjRepo", harborProjRepo)
```

Comment:
After a while you're going to end up with a fairly deeply-wrapped `ctx` object. I'm wondering if it would be better not to use the context to carry this parameter?